lib/Mojo/IOLoop/ReadWriteProcess.pm

```diff
@@ -477,18 +479,17 @@ sub start {
   $self->_status(undef);
   $self->session->enable;
 
-
-  if ($self->code) {
-    $self->_fork($self->code, @args);
-  }
-  elsif ($self->execute) {
-    $self->_open($self->execute, @args);
-  }
-
+  $self->session->_protect(sub {
+    if ($self->code) {
+      $self->_fork($self->code, @args);
+    }
+    elsif ($self->execute) {
+      $self->_open($self->execute, @args);
+    }
+    $self->session->register($self->pid() => $self);
+  });
   $self->write_pidfile;
```

Comment:
It's possible that at this point the process already exited, before `start` is emitted